server/models/procedures.sql

```diff
@@ -1463,4 +1463,115 @@ BEGIN
     sm.document_uuid = document_uuid
   GROUP BY i.uuid;
 END $$
+
+-- This processus inserts records relative to the stock movement integration in the posting journal
+CREATE PROCEDURE PostIntegration (
+  IN document_uuid BINARY(16),
+  IN date DATETIME,
+  IN enterprise_id SMALLINT(5) UNSIGNED,
+  IN project_id SMALLINT(5) UNSIGNED,
+  IN currency_id TINYINT(3) UNSIGNED,
+  IN user_id SMALLINT(5) UNSIGNED
+)
+BEGIN
+  DECLARE InvalidInventoryAccounts CONDITION FOR SQLSTATE '45006';
+  DECLARE current_fiscal_year_id MEDIUMINT(8) UNSIGNED;
+  DECLARE current_period_id MEDIUMINT(8) UNSIGNED;
+  DECLARE transaction_id VARCHAR(100);
+  DECLARE verify_invalid_accounts SMALLINT(5);
+  DECLARE STOCK_INTEGRATION_TRANSACTION_TYPE TINYINT(3) UNSIGNED;
+
+  -- should not be reassigned during the execution, to know why 12 please see the transaction_type table
+  SET STOCK_INTEGRATION_TRANSACTION_TYPE = 12;
+
+
+  -- getting the curent fiscal year
+  SET current_fiscal_year_id = (
+    SELECT id FROM fiscal_year AS fy
+    WHERE date BETWEEN fy.start_date AND DATE(ADDDATE(fy.start_date, INTERVAL fy.number_of_months MONTH)) AND fy.enterprise_id = enterprise_id
+  );
+
+  -- getting the period id
+  SET current_period_id = (
+    SELECT id FROM period AS p
+    WHERE DATE(date) BETWEEN DATE(p.start_date) AND DATE(p.end_date) AND p.fiscal_year_id = current_fiscal_year_id
+  );
+
+  CALL PostingJournalErrorHandler(enterprise_id, project_id, current_fiscal_year_id, current_period_id, 1, date);
+
+ -- Check that all every inventory has a stock account and a variation account - if they do not the transaction will be Unbalanced
+  SELECT 
+    COUNT(l.uuid) INTO verify_invalid_accounts
```

Comment:
This is a good idea!  I wonder if we can also do this on the client when we download the inventory for the first time...  That way someone doesn't spend a lot of time configuring lots to realize that their inventory is not set up properly...

Reply:
So what do you propose ?
A filter to fetch only inventories which have all accounts ? or a logic to check if a lot contains an inventory which has every account ?

Reply:
I'll make an issue about it and link it here.

Reply:
See https://github.com/IMA-WorldHealth/bhima-2.X/issues/2115.